### PR TITLE
feat: converted PreferencesController to typescript

### LINKED ISSUE #25923: Convert PreferencesController (implementation + tests) to TypeScript
### What is this about?

As a prerequisite for migrating `PreferencesController` to BaseController v2, and to support the TypeScript migration effort for the extension, we want to convert `PreferencesController` to TypeScript.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- All functions and methods are annotated with argument and return types.
- All properties and free variables are annotated with types.
- Relevant types from `preferences.d.ts` are folded into `preferences.ts`.
- `#` is used for private properties and methods instead of `_`.
- The test file is converted to TypeScript along with the implementation file.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

- https://github.com/MetaMask/metamask-extension/issues/23017

### PR DESCRIPTION
**Pull request by kanthesha**
## **Description**
As a prerequisite for migrating PreferencesController to BaseController v2, and to support the TypeScript migration effort for the extension, we want to convert PreferencesController to TypeScript.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26710?quickstart=1)

## **Related issues**

Fixes: #25923 

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## @cryptodev-2s **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

## @MajorLift  **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/preferences-controller.ts [hunk 1]
```diff
@@ -441,21 +534,21 @@ export default class PreferencesController {
...
    * @deprecated - Use setAccountName from the AccountsController
-   * @param {string} address - the account to set a label for
-   * @param {string} label - the custom label for the account
-   * @returns {Promise<string>}
+   * @param address - the account to set a label for
+   * @param label - the custom label for the account
+   * @returns
    */
-  async setAccountLabel(address, label) {
+  async setAccountLabel(address: string, label: string): Promise<string> {
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:551]: I believe this function shouldn't be async in first place, once removed the return type would be string

File: app/scripts/controllers/preferences-controller.ts [hunk 2]
```diff
@@ -498,11 +594,14 @@ export default class PreferencesController {
...
    *
-   * @param {string} preference - The preference to enable or disable.
-   * @param {boolean |object} value - Indicates whether or not the preference should be enabled or disabled.
-   * @returns {Promise<object>} Promises a new object; the updated preferences object.
+   * @param preference - The preference to enable or disable.
+   * @param value - Indicates whether or not the preference should be enabled or disabled.
+   * @returns Promises a updated Preferences object.
    */
-  async setPreference(preference, value) {
+  async setPreference(
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:601]: I believe this function shouldn't be async in first place, once removed the return type would be Preferences

File: app/scripts/controllers/preferences-controller.ts [hunk 3]
```diff
@@ -516,47 +615,49 @@ export default class PreferencesController {
...
   /**
    * A setter for the `ipfsGateway` property
    *
-   * @param {string} domain - The new IPFS gateway domain
-   * @returns {Promise<string>} A promise of the update IPFS gateway domain
+   * @param domain - The new IPFS gateway domain
+   * @returns A promise of the update IPFS gateway domain
    */
-  async setIpfsGateway(domain) {
+  async setIpfsGateway(domain: string): Promise<string> {
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:639]: I believe this function shouldn't be async in first place, once removed the return type would be string

File: app/scripts/controllers/preferences-controller.ts [hunk 4]
```diff
@@ -516,47 +615,49 @@ export default class PreferencesController {
...
   }
 
   /**
    * A setter for the `isIpfsGatewayEnabled` property
    *
-   * @param {boolean} enabled - Whether or not IPFS is enabled
+   * @param enabled - Whether or not IPFS is enabled
    */
-  async setIsIpfsGatewayEnabled(enabled) {
+  async setIsIpfsGatewayEnabled(enabled: boolean): Promise<void> {
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:649]: I believe this function shouldn't be async in first place, once removed the return type would be void

File: app/scripts/controllers/preferences-controller.ts [hunk 5]
```diff
@@ -565,20 +666,24 @@ export default class PreferencesController {
...
   }
 
   /**
    * A setter for the user preference to dismiss the seed phrase backup reminder
    *
-   * @param {bool} dismissSeedBackUpReminder - User preference for dismissing the back up reminder.
+   * @param dismissSeedBackUpReminder - User preference for dismissing the back up reminder.
    */
-  async setDismissSeedBackUpReminder(dismissSeedBackUpReminder) {
+  async setDismissSeedBackUpReminder(
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:684]: I believe this function shouldn't be async in first place, once removed the return type would be void
also this.store.updateState is not a promise no need to await

File: app/scripts/controllers/preferences-controller.ts [hunk 6]
```diff
@@ -17,11 +25,90 @@ const testNetworks = {
...
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private network: any;
+
   /**
    *
-   * @typedef {object} PreferencesController
-   * @param {object} opts - Overrides the defaults for the initial state of this.store
+   * @param opts - Overrides the defaults for the initial state of this.store
    * @property {object} messenger - The controller messenger
    * @property {object} store The stored object containing a users preferences, stored in local storage
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:113]: Minor suggestion:
How about renaming this store property to match the actual property name, `initState`?
Additionally, I believe we can remove the `{object}` type annotation now that we have a defined options type.

File: app/scripts/controllers/preferences-controller.ts [hunk 7]
```diff
@@ -17,11 +25,90 @@ const testNetworks = {
   [CHAIN_IDS.LINEA_SEPOLIA]: true,
 };
 
+type PreferencesControllerOptions = {
+  networkConfigurations?: Record<string, { chainId: Hex }>;
+  initState?: Partial<PreferencesControllerState>;
+  initLangCode?: string;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  network?: any;
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:34]: I believe the `network` property can just be removed I don't see it used anywhere.

File: app/scripts/controllers/preferences-controller.ts [hunk 8]
```diff
@@ -17,11 +25,90 @@ const testNetworks = {
...
 export default class PreferencesController {
+  store: ObservableStore<PreferencesControllerState>;
+
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private messagingSystem: any;
+
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private network: any;
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:107]: Same as the below comment if we remove the network property from the constructor options. we can remove this private variable

File: app/scripts/controllers/preferences-controller.ts [hunk 9]
```diff
@@ -17,11 +25,90 @@ const testNetworks = {
...
+  networkConfigurations?: Record<string, { chainId: Hex }>;
+  initState?: Partial<PreferencesControllerState>;
+  initLangCode?: string;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  network?: any;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  messenger?: any;
+};
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:38]:
```suggestion

const controllerName = 'PreferencesController';

/**
 * Returns the state of the {@link PreferencesController}.
 */
export type PreferencesControllerGetStateAction = {
  type: 'PreferencesController:getState';
  handler: () => PreferencesControllerState;
};

/**
 * Actions exposed by the {@link PreferencesController}.
 */
export type PreferencesControllerActions = PreferencesControllerGetStateAction;

/**
 * Event emitted when the state of the {@link PreferencesController} changes.
 */
export type PreferencesControllerStateChangeEvent = {
  type: 'PreferencesController:stateChange';
  payload: [PreferencesControllerState, []];
};

/**
 * Events emitted by {@link PreferencesController}.
 */
export type PreferencesControllerEvents = PreferencesControllerStateChangeEvent;

/**
 * Actions that this controller is allowed to call.
 */
export type AllowedActions =
  | AccountsControllerGetAccountByAddressAction
  | AccountsControllerSetAccountNameAction
  | AccountsControllerGetSelectedAccountAction
  | AccountsControllerSetSelectedAccountAction;

/**
 * Events that this controller is allowed to subscribe.
 */
export type AllowedEvents = AccountsControllerChangeEvent;

export type PreferencesControllerMessenger = RestrictedControllerMessenger<
  typeof controllerName,
  PreferencesControllerActions | AllowedActions,
  PreferencesControllerEvents | AllowedEvents,
  AllowedActions['type'],
  AllowedEvents['type']
>;

type PreferencesControllerOptions = {
  networkConfigurations?: Record<string, { chainId: Hex }>;
  initState?: Partial<PreferencesControllerState>;
  initLangCode?: string;
  messenger: PreferencesControllerMessenger;
};
```

I believe the network property can just be removed I don't see it used anywhere.

Comment (cryptodev-2s) [reply app/scripts/controllers/preferences-controller.ts:38]: I believe the`network` property can just be removed I don't see it used anywhere.

File: app/scripts/controllers/preferences-controller.ts [hunk 10]
```diff
@@ -17,11 +25,90 @@ const testNetworks = {
...
+  useExternalServices: boolean;
+  textDirection?: string;
+};
+
 export default class PreferencesController {
+  store: ObservableStore<PreferencesControllerState>;
+
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private messagingSystem: any;
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:103]:
```suggestion
private messagingSystem: PreferencesControllerMessenger;
```

File: app/scripts/controllers/preferences-controller.ts [hunk 11]
```diff
@@ -140,67 +228,66 @@ export default class PreferencesController {
       this.#handleAccountsControllerSync.bind(this),
     );
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.ts:229]:
```suggestion
    this.messagingSystem = opts.messenger;
    this.messagingSystem.registerActionHandler(
      `PreferencesController:getState`,
      () => this.store.getState(),
    );
    this.messagingSystem.registerInitialEventPayload({
      eventType: `PreferencesController:stateChange`,
      getPayload: () => [this.store.getState(), []],
    });

    this.messagingSystem.subscribe(
      'AccountsController:stateChange',
      this.#handleAccountsControllerSync.bind(this),
    );
````

File: app/scripts/controllers/preferences-controller.ts (hunk 1 above)
Comment (cryptodev-2s) [reply app/scripts/controllers/preferences-controller.ts:551]: Nit:
We could also consider updating the function by switching the order to check the address first. Additionally, it should throw an error if the account is not found to ensure that `account.id` passes TypeScript validation, as `account` could be undefined.
````
 setAccountLabel(address: string, label: string): string {
    if (!address) {
      throw new Error(
        `setAccountLabel requires a valid address, got ${String(address)}`,
      );
    }

    const account = this.messagingSystem.call(
      'AccountsController:getAccountByAddress',
      address,
    );

    if (!account) {
      throw new Error(`Account for '${address} not found`);
    }

    this.messagingSystem.call(
      'AccountsController:setAccountName',
      account.id,
      label,
    );

    return label;
  }
```

File: app/scripts/controllers/preferences-controller.test.ts [hunk 12]
```diff
@@ -2,35 +2,34 @@
...
 ).networkConfigurations;
 
 describe('preferences controller', () => {
-  let controllerMessenger;
-  let preferencesController;
-  let accountsController;
-  let tokenListController;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let controllerMessenger: any;
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.test.ts:30]:
```suggestion
 let controllerMessenger: ControllerMessenger<
    | PreferencesControllerActions
    | AllowedActions
    | KeyringControllerGetAccountsAction
    | KeyringControllerGetKeyringsByTypeAction
    | KeyringControllerGetKeyringForAccountAction,
    | PreferencesControllerEvents
    | KeyringControllerStateChangeEvent
    | KeyringControllerAccountRemovedEvent
    | SnapControllerStateChangeEvent
    | AllowedEvents
  >;
```

File: app/scripts/controllers/preferences-controller.test.ts [hunk 13]
```diff
@@ -357,7 +349,6 @@ describe('preferences controller', () => {
     it('should keep initial value of useTokenDetection for existing users', function () {
       const preferencesControllerExistingUser = new PreferencesController({
```
Comment (cryptodev-2s) [inline app/scripts/controllers/preferences-controller.test.ts:350]:
```suggestion
    it('should keep initial value of useTokenDetection for existing users', function () {
      // TODO: Remove unregisterActionHandler and clearEventSubscriptions once the PreferencesController has been refactored to use the withController pattern.
      controllerMessenger.unregisterActionHandler(
        'PreferencesController:getState',
      );
      controllerMessenger.clearEventSubscriptions(
        'PreferencesController:stateChange',
      );
      const preferencesControllerExistingUser = new PreferencesController({
        messenger: controllerMessenger.getRestricted({
          name: 'PreferencesController',
          allowedActions: [],
          allowedEvents: ['AccountsController:stateChange'],
        }),
        initLangCode: 'en_US',
        initState: {
          useTokenDetection: false,
        },
        networkConfigurations: NETWORK_CONFIGURATION_DATA,
      });
      const state = preferencesControllerExistingUser.store.getState();
      expect(state.useTokenDetection).toStrictEqual(false);
    });
  });

**Comment by cryptodev-2s** [comment]
Nit:
We can also update the Controller initialisation in metamask-controller.js to
```
   const preferencesMessenger = this.controllerMessenger.getRestricted({
      name: 'PreferencesController',
      allowedActions: [
        'AccountsController:setSelectedAccount',
        'AccountsController:getAccountByAddress',
        'AccountsController:setAccountName',
        'AccountsController:setSelectedAccount',
      ],
      allowedEvents: ['AccountsController:stateChange'],
    });

    this.preferencesController = new PreferencesController({
      initState: initState.PreferencesController,
      initLangCode: opts.initLangCode,
      messenger: preferencesMessenger,
      networkConfigurations: this.networkController.state.networkConfigurations,
    });

**Review by cryptodev-2s — DISMISSED** [review]

File: app/scripts/controllers/preferences-controller.ts [hunk 14]
```diff
@@ -17,32 +33,150 @@ const testNetworks = {
...
+export type AllowedActions =
+  | AccountsControllerGetAccountByAddressAction
+  | AccountsControllerSetAccountNameAction
+  | AccountsControllerGetSelectedAccountAction
+  | AccountsControllerSetSelectedAccountAction;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents = AccountsControllerChangeEvent;
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:76]:
```suggestion
export type AllowedEvents = AccountsControllerChangeEvent;

```

File: app/scripts/controllers/preferences-controller.ts [hunk 15]
```diff
@@ -17,32 +33,150 @@ const testNetworks = {
...
-    const addedNonMainNetwork = Object.values(
-      opts.networkConfigurations,
-    ).reduce((acc, element) => {
+   * @property initState.knownMethodData Contains all data methods known by the user
+   * @property initState.currentLocale The preferred language locale key
+   * @property initState.selectedAddress A hex string that matches the currently selected address in the app
+   */
+  constructor(opts: PreferencesControllerOptions) {
+    const addedNonMainNetwork: Record<Hex, boolean> = Object.values(
+      opts.networkConfigurations || {},
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:172]:
```suggestion
      opts.networkConfigurations ?? {},
```

File: app/scripts/controllers/preferences-controller.ts [hunk 16]
```diff
@@ -77,7 +211,7 @@ export default class PreferencesController {
         ...testNetworks,
       },
       knownMethodData: {},
-      currentLocale: opts.initLangCode,
+      currentLocale: opts.initLangCode || '',
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:214]:
```suggestion
      currentLocale: opts.initLangCode ?? '',
```

**Review by MajorLift — COMMENTED** [review]
nits

File: app/scripts/controllers/preferences-controller.ts [hunk 17]
```diff
@@ -17,32 +33,150 @@ const testNetworks = {
...
+>;
+
+type PreferencesControllerOptions = {
+  networkConfigurations?: Record<string, { chainId: Hex }>;
+  initState?: Partial<PreferencesControllerState>;
+  initLangCode?: string;
+  messenger: PreferencesControllerMessenger;
+};
+
+type Preferences = {
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:92]:
```suggestion
export type Preferences = {
```

File: app/scripts/controllers/preferences-controller.ts [hunk 18]
```diff
@@ -120,87 +254,84 @@ export default class PreferencesController {
...
     });
 
-    this.messagingSystem?.subscribe(
+    this.messagingSystem.subscribe(
       'AccountsController:stateChange',
       this.#handleAccountsControllerSync.bind(this),
     );
 
-    global.setPreference = (key, value) => {
+    globalThis.setPreference = (key: string, value: boolean) => {
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:275]:
```suggestion
    globalThis.setPreference = (key: keyof Preferences, value: boolean) => {
```

File: app/scripts/controllers/preferences-controller.ts [hunk 19]
```diff
@@ -498,11 +639,11 @@ export default class PreferencesController {
...
    *
-   * @param {string} preference - The preference to enable or disable.
-   * @param {boolean |object} value - Indicates whether or not the preference should be enabled or disabled.
-   * @returns {Promise<object>} Promises a new object; the updated preferences object.
+   * @param preference - The preference to enable or disable.
+   * @param value - Indicates whether or not the preference should be enabled or disabled.
+   * @returns Promises a updated Preferences object.
    */
-  async setPreference(preference, value) {
+  setPreference(preference: string, value: boolean | object): Preferences {
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:646]:
```suggestion
  setPreference(
    preference: keyof Preferences,
    value: Preferences[typeof preference],
  ): Preferences {
```

File: types/global.d.ts [hunk 20]
```diff
@@ -278,4 +278,6 @@ export declare global {
    * Unions T with U; U's properties will override T's properties
    */
   type OverridingUnion<T, U> = Omit<T, keyof U> & U;
+
+  function setPreference(key: string, value: boolean);
```
Comment (MajorLift) [inline types/global.d.ts:282]:
```suggestion
  function setPreference(key: keyof Preferences, value: boolean);
```

```diff
diff --git a/types/global.d.ts b/types/global.d.ts
index 33e5c52fd0..5b963f1ec5 100644
--- a/types/global.d.ts
+++ b/types/global.d.ts
@@ -21,6 +21,7 @@ import {
   OffscreenCommunicationTarget,
   TrezorAction,
 } from 'shared/constants/offscreen-communication';
+import type { Preferences } from '../app/scripts/controllers/preferences-controller';
```

**Review by MajorLift — DISMISSED** [review]
Narrow references to properties of the `Preferences` type: from `string` to `keyof Preferences`

File: app/scripts/controllers/preferences-controller.ts [hunk 21]
```diff
@@ -17,32 +33,150 @@ const testNetworks = {
...
+  theme: ThemeType;
+  snapsAddSnapAccountModalDismissed: boolean;
+  useExternalNameSources: boolean;
+  useTransactionSimulations: boolean;
+  enableMV3TimestampSave: boolean;
+  useExternalServices: boolean;
+  textDirection?: string;
+};
+
 export default class PreferencesController {
```
Comment (MajorLift) [inline app/scripts/controllers/preferences-controller.ts:150]: nit: What do you think about aligning this export statement with the core controller here? The extra diffs for going back and forth between default vs. named import seem unnecessary.

```suggestion
export class PreferencesController {
```

```diff
diff --git a/app/scripts/controllers/mmi-controller.ts b/app/scripts/controllers/mmi-controller.ts
index 7cc024d0cd..4b1e534907 100644
--- a/app/scripts/controllers/mmi-controller.ts
+++ b/app/scripts/controllers/mmi-controller.ts
@@ -41,7 +41,7 @@ import AccountTracker from '../lib/account-tracker';
 import { getCurrentChainId } from '../../../ui/selectors';
 import MetaMetricsController from './metametrics';
 import { getPermissionBackgroundApiMethods } from './permissions';
-import PreferencesController from './preferences-controller';
+import { PreferencesController } from './preferences-controller';
 import { AppStateController } from './app-state';
 
 type UpdateCustodianTransactionsParameters = {
diff --git a/app/scripts/controllers/preferences-controller.test.ts b/app/scripts/controllers/preferences-controller.test.ts
index 56885232be..1822071e22 100644
--- a/app/scripts/controllers/preferences-controller.test.ts
+++ b/app/scripts/controllers/preferences-controller.test.ts
@@ -14,12 +14,13 @@ import { SnapControllerStateChangeEvent } from '@metamask/snaps-controllers';
 import { CHAIN_IDS } from '../../../shared/constants/network';
 import { mockNetworkState } from '../../../test/stub/networks';
 import { ThemeType } from '../../../shared/constants/preferences';
-import PreferencesController, {
+import type {
   AllowedActions,
   AllowedEvents,
   PreferencesControllerActions,
   PreferencesControllerEvents,
 } from './preferences-controller';
+import { PreferencesController } from './preferences-controller';
 
 const NETWORK_CONFIGURATION_DATA = mockNetworkState(
   {
diff --git a/app/scripts/lib/ppom/ppom-middleware.ts b/app/scripts/lib/ppom/ppom-middleware.ts
index d623d4868a..d3cbea90f3 100644
--- a/app/scripts/lib/ppom/ppom-middleware.ts
+++ b/app/scripts/lib/ppom/ppom-middleware.ts
@@ -11,7 +11,7 @@ import { detectSIWE } from '@metamask/controller-utils';
 
 import { MESSAGE_TYPE } from '../../../../shared/constants/app';
 import { SIGNING_METHODS } from '../../../../shared/constants/transaction';
-import PreferencesController from '../../controllers/preferences-controller';
+import { PreferencesController } from '../../controllers/preferences-controller';
 import { AppStateController } from '../../controllers/app-state';
 import { LOADING_SECURITY_ALERT_RESPONSE } from '../../../../shared/constants/security-provider';
 import { getProviderConfig } from '../../../../ui/ducks/metamask/metamask';
diff --git a/app/scripts/metamask-controller.js b/app/scripts/metamask-controller.js
index b7214ad198..478e0cc750 100644
--- a/app/scripts/metamask-controller.js
+++ b/app/scripts/metamask-controller.js
@@ -283,7 +283,7 @@ import { NetworkOrderController } from './controllers/network-order';
 import { AccountOrderController } from './controllers/account-order';
 import createOnboardingMiddleware from './lib/createOnboardingMiddleware';
 import { isStreamWritable, setupMultiplex } from './lib/stream-utils';
-import PreferencesController from './controllers/preferences-controller';
+import { PreferencesController } from './controllers/preferences-controller';
 import AppStateController from './controllers/app-state';
 import AlertController from './controllers/alert';
 import OnboardingController from './controllers/onboarding';
diff --git a/shared/constants/mmi-controller.ts b/shared/constants/mmi-controller.ts
index 67123b83d5..0efcceeb08 100644
--- a/shared/constants/mmi-controller.ts
+++ b/shared/constants/mmi-controller.ts
@@ -4,7 +4,7 @@ import { TransactionUpdateController } from '@metamask-institutional/transaction
 import { CustodyController } from '@metamask-institutional/custody-controller';
 import { SignatureController } from '@metamask/signature-controller';
 import { NetworkController } from '@metamask/network-controller';
-import PreferencesController from '../../app/scripts/controllers/preferences-controller';
+import { PreferencesController } from '../../app/scripts/controllers/preferences-controller';
 import { AppStateController } from '../../app/scripts/controllers/app-state';
 import AccountTracker from '../../app/scripts/lib/account-tracker';
 import MetaMetricsController from '../../app/scripts/controllers/metametrics';
```

**Review by cryptodev-2s — APPROVED** [review]

**Review by MajorLift — APPROVED** [review]
LGTM

**Review by desi — COMMENTED** [review]
@kanthesha can you please check the pre merge checklist and make sure you have checked against all of those. @MajorLift and @cryptodev-2s did either of you do testing? If so can you indicate such in the reviewer checklist?

**Review by desi — APPROVED** [review]
Based on previous reviewers indicating testing and complete review I am approving in order to allow merging (code owners)

**Pull request merged**